intercom-common/src/tyhandlers.rs

```diff
@@ -93,80 +104,91 @@ impl TypeHandler for ComItfParam {
 }
 
 /// String parameter handler. Converts between Rust String and COM BSTR types.
-struct StringParam( Type );
+struct StringParam { ty: Type, context: TypeContext }
```

Comment:
No. We don't have `BSTR` and `non-BSTR` parameters. The `TyHandler` is static part of the library graph. I don't really want to generate duplicate method declarations for the logical library model for different return values.

Although writing this; I just realized we can't use the `TypeContext` as something that specifies Automation interfaces vs Raw interfaces either for this smae reason. As long as the `TypeContext` is part of the Param itself, it needs to depend only on the static parts of the model and not dynamic parts such as "what kind of interfaces we are generating at this moment".

Reply:
In the future we will have "U8String" parameter which will avoid string conversions in the interface. 

Should we represent these as sub-types of StringParam or is the StringParam the default type for strings and then we will just have U8StringParam struct for the U8String?

Reply:
Why `U8String` as opposed to [`CString`](https://doc.rust-lang.org/std/ffi/struct.CString.html)?

The concept of "C-string" defines not only the data type but also the zero termination.

I believe we should represent these as the StringParam. The Automation vs C/Raw/U8 string difference is carried in the context and affects the COM representation (BString/BStr vs CString/CStr) - everything else stays the same and is covered by type inference and implemented traits. I don't see the need to duplicate the whole StringParams just to change the internal string type name.

Reply:
The U8 string will need to include the number bytes in the buffer so that in can be converted into &str without the need to count the number of bytes in the buffer.

I want basically want to make it a string type that does not require any additional operations when passed through the interface. That will allow developers to decide when to take the penalty of converting their own strings into a format that is compatible with Rust COM interface.